frontend/app/.server/routes/mappers/benefit-renewal.state.mapper.ts

```diff
@@ -422,32 +397,28 @@ export class DefaultBenefitRenewalStateMapper implements BenefitRenewalStateMapp
     };
   }
 
-  private toDentalBenefits({ existingDentalBenefits, hasFederalBenefitsChanged, hasProvincialTerritorialBenefitsChanged, renewedDentalBenefits }: ToDentalBenefitsArgs) {
-    if (!hasFederalBenefitsChanged && !hasProvincialTerritorialBenefitsChanged) return existingDentalBenefits;
+  private toDentalBenefits({ existingDentalBenefits, hasFederalProvincialTerritorialBenefitsChanged, renewedDentalBenefits }: ToDentalBenefitsArgs) {
+    if (!hasFederalProvincialTerritorialBenefitsChanged) return existingDentalBenefits;
     if (!renewedDentalBenefits) return [];
 
     const dentalBenefits = [];
 
-    if (!hasFederalBenefitsChanged) {
+    if (renewedDentalBenefits.hasFederalBenefits && renewedDentalBenefits.federalSocialProgram && !validator.isEmpty(renewedDentalBenefits.federalSocialProgram)) {
+      dentalBenefits.push(renewedDentalBenefits.federalSocialProgram);
+    } else {
       const federalGovernmentInsurancePlans = this.federalGovernmentInsurancePlanService.listFederalGovernmentInsurancePlans();
-      const existingFederalGovernmentInsurancePlan = federalGovernmentInsurancePlans
-        .filter((federalGovernmentInsurancePlan) => existingDentalBenefits.includes(federalGovernmentInsurancePlan.id))
-        .map((federalGovernmentInsurancePlan) => federalGovernmentInsurancePlan.id);
+      const existingFederalGovernmentInsurancePlan = federalGovernmentInsurancePlans.filter((plan) => existingDentalBenefits.includes(plan.id)).map((plan) => plan.id);
 
       dentalBenefits.push(...existingFederalGovernmentInsurancePlan);
-    } else if (renewedDentalBenefits.hasFederalBenefits && renewedDentalBenefits.federalSocialProgram && !validator.isEmpty(renewedDentalBenefits.federalSocialProgram)) {
-      dentalBenefits.push(renewedDentalBenefits.federalSocialProgram);
     }
 
-    if (!hasProvincialTerritorialBenefitsChanged) {
+    if (renewedDentalBenefits.hasProvincialTerritorialBenefits && renewedDentalBenefits.provincialTerritorialSocialProgram && !validator.isEmpty(renewedDentalBenefits.provincialTerritorialSocialProgram)) {
+      dentalBenefits.push(renewedDentalBenefits.provincialTerritorialSocialProgram);
+    } else {
       const provincialGovernmentInsurancePlans = this.provincialGovernmentInsurancePlanService.listProvincialGovernmentInsurancePlans();
-      const existingProvincialGovernmentInsurancePlan = provincialGovernmentInsurancePlans
-        .filter((provincialGovernmentInsurancePlan) => existingDentalBenefits.includes(provincialGovernmentInsurancePlan.id))
-        .map((provincialGovernmentInsurancePlan) => provincialGovernmentInsurancePlan.id);
+      const existingProvincialGovernmentInsurancePlan = provincialGovernmentInsurancePlans.filter((plan) => existingDentalBenefits.includes(plan.id)).map((plan) => plan.id);
```

Comment:
There's no need for the `else` blocks in this method.

Federal and provincial/territorial benefits are updated simultaneously when the user submits the form, so we intend to overwrite all existing benefit information for the applicant.

Keeping the `else` blocks would result in unintended behavior. If the applicant selects "No" for having federal or provincial/territorial benefits, their existing information on file would still be used, potentially including outdated benefits.